docs/swagger_v2.yml

```diff
@@ -0,0 +1,663 @@
+        format: date-time
+        description: Most recent time that app was updated. Always in UTC.
+        readOnly: true
+
+  Fn:
+    type: object
+    properties:
+      id:
+        type: string
+        description: Unique identifier
+        readOnly: true
+      name:
+        type: string
+        description: unique name for this function.
+      app_id:
+        type: string
+        description: App ID.
+      image:
+        type: string
+        description: "full container image name, e.g. hub.docker.com/fnproject/yo or fnproject/yo (default registry: hub.docker.com)"
+      mem:
+        type: integer
+        format: uint64
+        description: Max usable memory given to function (MiB).
+      cpu:
+        type: string
+        description: Max usable CPU cores for this trigger. Value in MilliCPUs (eg. 500m) or as floating-point (eg. 0.5)
+      timeout:
+        type: integer
+        default: 30
+        format: int32
+        description: Timeout for executions of a function. Value in Seconds
+      idle_timeout:
+        type: integer
+        default: 30
+        format: int32
+        description: Hot functions idle timeout before container termination. Value in Seconds
+      config:
+        type: object
+        description: Func configuration key values
+        additionalProperties:
+          type: string
+      format:
+        enum:
+          - default
+          - http
+          - json
+          - cloudevent
+        description: Payload format sent into function.
+        type: string
+      annotations:
+        type: object
+        description: Func annotations - this is a map of annotations attached to this func, keys must not exceed 128 bytes and must consist of non-whitespace printable ascii characters, and the seralized representation of individual values must not exeed 512 bytes
+        additionalProperties:
+          type: object
+      created_at:
+        type: string
+        format: date-time
+        description: Time when function was created. Always in UTC RFC3339.
+        readOnly: true
+      updated_at:
+        type: string
+        format: date-time
+        description: Most recent time that function was updated. Always in UTC RFC3339.
+        readOnly: true
+
+  FnList:
+    type: object
+    required:
+      - items
+    properties:
+      next_cursor:
+        type: string
+        description: cursor to send with subsequent request to receive the next page, if non-empty
+        readOnly: true
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Fn'
+
+  Trigger:
+    type: object
+    properties:
+      id:
+        type: string
+        description: Unique trigger identifier
+        readOnly: true
+      name:
+        type: string
+        description: unique name for this trigger, used to identify this trigger
+      type:
+        type: string
+        description: class of trigger, e.g. schedule, http
+        enum: [http, stream]
```

Comment:
string here? an extension could limit the domain here but i'm not sure in core that we should do it.